scaladoc/src/dotty/tools/scaladoc/site/helpers/ConfigLoader.scala

```diff
@@ -0,0 +1,108 @@
+    data.get(field) match {
+      case Some(value: mutable.LinkedHashMap[_, _]) =>
+        new Config(value.asInstanceOf[mutable.LinkedHashMap[String, Any]])
+      case Some(value: java.util.ArrayList[_]) =>
+        value.asScala.toSeq.map {
+          case map: java.util.Map[_, _] =>
+            new Config(
+              map
+                .asInstanceOf[java.util.Map[String, Any]]
+                .asScala
+                .to(mutable.LinkedHashMap)
+            )
+          case other => other
+        }
+      case Some(value) => value
+      case None => throw new NoSuchElementException(s"No such element: $field")
+    }
+  }
+
+  def get[T](key: String): Option[T] = {
+    data.get(key).map(_.asInstanceOf[T])
+  }
+
+  //  method to convert the internal map to a Java map
+  def convertToJava: java.util.Map[String, Any] = {
+    data.map {
+      case (key, value: Config) => key -> value.convertToJava
+      case (key, value: mutable.LinkedHashMap[_, _]) =>
+        key -> value.asInstanceOf[mutable.LinkedHashMap[String, Any]].asJava
+      case (key, value) => key -> value
+    }.asJava
+  }
+}
+
+class ConfigLoader {
+  def loadConfig(basePath: String): Config = {
+    val configMap = mutable.LinkedHashMap[String, Any]()
+    val yamlFileNames = Seq("_config.yml", "_config.yaml")
+
+    try {
+      val yaml = new Yaml()
+      val baseDir = new File(basePath).getAbsolutePath
+
+      val configFile = yamlFileNames
+        .map(baseDir + File.separator + _)
+        .map(Paths.get(_))
+        .find(Files.exists(_))
+
+      configFile match {
+        case Some(path) =>
+          val inputStream: InputStream = Files.newInputStream(path)
+          val data = yaml.load[java.util.Map[String, Any]](inputStream)
+          configMap ++= data.asScala.toMap
+
+          // Check for language folders
+          val languagesOpt =
+            configMap.get("languages").collect { case list: java.util.List[_] =>
+              list.asScala.toList.collect { case map: java.util.Map[_, _] =>
+                map.asScala.toMap.collect { case (key: String, value: String) =>
+                  key -> value
+                }
+              }
+            }
+
+        // languagesOpt match {
+        //   case Some(languages) =>
+        //     for (language <- languages) {
+        //       // Cast the key to String to avoid type mismatch errors
+        //       val languageCode = language("code".asInstanceOf[language.K]).asInstanceOf[String]
+        //       val languageFolderPath = Paths.get(baseDir, languageCode)
+        //       if (!Files.exists(languageFolderPath) || !Files.isDirectory(languageFolderPath)) {
+        //         throw new IllegalArgumentException(s"Language folder for '$languageCode' does not exist at path: $languageFolderPath")
+        //       }
+        //     }
+        //   case None =>
+        //     println(s"Warning: No languages found in configuration.")
+        // }
```

Comment:
what happened here?

Reply:
i commented this out as it was only checking for the existence of folder, but i figured out we need a way to check the files as well